src/tests/data-test/apod.datasource.test.ts

```diff
@@ -92,4 +92,24 @@ describe("Apod datasource ", async () => {
     expect(saved!.url).to.be.eq(apod.url);
     expect(saved!.mediaType).to.be.eq(apod.mediaType);
   });
+
+  it("Should get a random apod with title", async () => {
+    let apod = await apodDatasource.getRandomApod();
+    expect(apod.title).to.exist;
+  });
+
+  it("Should get a random apod with url", async () => {
+    let apod = await apodDatasource.getRandomApod();
+    expect(apod.url).to.exist;
+  });
+
+  it("Should get a random apod with explanation", async () => {
+    let apod = await apodDatasource.getRandomApod();
+    expect(apod.explanation).to.exist;
+  });
+
+  it("Should get a random apod with media type", async () => {
+    let apod = await apodDatasource.getRandomApod();
+    expect(apod.mediaType).to.exist;
+  });
```

Comment:
Vc ta batendo na API da NASA pra fazer esses testes né? Não é legal fazer assim porque se um dia a API da NASA falhar seu teste vai falhar, mas não por erro no código e sim por uma interferência externa que não controlamos. 

Em casos assim a gente costuma "stubar" o retorno do `getRandomApod`, isto é, simular a situação que queremos. Você pode fazer isso com a biblitoeca [sinon e seu recurso de stub](https://sinonjs.org/releases/latest/stubs/)

Qualquer coisa depois a gente fala melhor sobre isso.

Reply:
Vamos pendurar isso num to do.